clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -264,6 +264,63 @@ public void testReadBytes() {
         this.subTest(buffer);
     }
 
+    /**
+     * Test to read a simple file as string.
+     */
+    @Test
+    public void testFileAsStringSimpleFile() throws IOException {
+        File tempFile = File.createTempFile("UtilsTestTempFlie", ".tmp");
+        try {
+            String testContent = "Test Content";
+            Files.write(tempFile.toPath(), testContent.getBytes());
+            assertEquals(testContent, Utils.readFileAsString(tempFile.getPath()));
+        } finally {
+            Files.deleteIfExists(tempFile.toPath());
+        }
+    }
+
+    /**
+     * Test to read content of named pipe as string.
+     */
+    @Test
+    public void testFileAsStringNamedPipe() throws Exception {
+
+        // Create a temporary name for named pipe
+        Random random = new Random();
+        long n = random.nextLong();
+        n = n == Long.MIN_VALUE ? 0 : Math.abs(n);
+
+        // Use the name to create a FIFO in tmp directory
+        String tmpDir = System.getProperty("java.io.tmpdir");
+        String fifoName = "fifo-" + n + ".tmp";
+        File fifo = new File(tmpDir, fifoName);
```

Comment:
Is there a reason why we have all this naming logic?

Reply:
Need to create a temporary name w/o creating the file on file system (as that will be created by `mkfifo` command). I couldn't find an existing utility to do this.

Another option will be to create a temp file, then delete it and then use that name. Let me know what you think.